src/nanovg_pick.c

```diff
@@ -0,0 +1,1651 @@
+
+	pp->id = id;
+					
+	while (i < ncommands) {
+		int cmd = (int)commands[i];
+		
+		switch (cmd) {
+		case NVG_MOVETO:
+			start[0] = commands[i+1];
+			start[1] = commands[i+2];
+
+			// Start a new path for each sub path to handle sub paths that
+			// intersect other sub paths.
+			prev = psp;
+			psp = nvg__allocPickSubPath(ps);
+			if (psp == NULL) { psp = prev; break; }
+			psp->firstSegment = -1;
+			psp->winding = NVG_SOLID;
+			psp->next = prev;
+
+			nvg__pickSceneAddPoints(ps, &commands[i+1], 1);
+			i += 3;
+			break;
+		case NVG_LINETO:
+			firstPoint = nvg__pickSceneAddPoints(ps, &commands[i+1], 1);
+			nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, cmd, NVG_PICK_CORNER);
+			i += 3;
+			break;
+		case NVG_BEZIERTO:
+			// Split the curve at it's dx==0 or dy==0 inflection points.
+			// Thus:
+			//		A horizontal line only ever interects the curves once.
+			//	and
+			//		Finding the closest point on any curve converges more reliably.
+			
+			// NOTE: We could just split on dy==0 here.
+						
+			memcpy(&points[0], &ps->points[(ps->npoints - 1) * 2], sizeof(float) * 2);
+			memcpy(&points[2], &commands[i+1], sizeof(float) * 2 * 3);
+							
+			nvg__bezierInflections(points, 1, &ninflections, inflections);
+			nvg__bezierInflections(points, 0, &ninflections, inflections);
+			
+			if (ninflections) {
+				nvg__smallsort(inflections, ninflections);
+
+				float previnfl = 0;
+				float t;
+				
+				float pointsA[8], pointsB[8];
+
+				int infl;
+				for (infl = 0; infl < ninflections; ++infl) {
+					if (nvg__absf(inflections[infl] - previnfl) < NVG_PICK_EPS)
+						continue;
+						
+					t = (inflections[infl] - previnfl) * (1.0f / (1.0f - previnfl));
+					
+					previnfl = inflections[infl];
+						
+					nvg__splitBezier(points, t, pointsA, pointsB);
+
+					firstPoint = nvg__pickSceneAddPoints(ps, &pointsA[2], 3);
+					nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, cmd, (infl == 0) ? NVG_PICK_CORNER : 0);
+
+					memcpy(points, pointsB, sizeof(float) * 8);												
+				}
+				
+				firstPoint = nvg__pickSceneAddPoints(ps, &pointsB[2], 3);
+				nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, cmd, 0);
+			} else {
+				firstPoint = nvg__pickSceneAddPoints(ps, &commands[i+1], 3);
+				nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, cmd, NVG_PICK_CORNER);
+			}
+			i += 7;
+			break;
+		case NVG_CLOSE:
+			if (ps->points[(ps->npoints - 1) * 2] != start[0] ||
+				ps->points[(ps->npoints - 1) * 2 + 1] != start[1]) {
+				firstPoint = nvg__pickSceneAddPoints(ps, start, 1);
+				nvg__pickSubPathAddSegment(ps, psp, firstPoint - 1, NVG_LINETO, NVG_PICK_CORNER);
+			}
+			psp->closed = 1;
+			
+			i++;
+			break;
+		case NVG_WINDING:
+			psp->winding = (short)(int)commands[i+1];
+			if (psp->winding == NVG_HOLE)
+				hasHoles = 1;
+			i += 2;
+			break;
+		default:
+			i++;
+			break;
+		}
+	}
+	
+	pp->flags = (hasHoles ? NVG_PICK_HASHOLES : 0) | (forStroke ? NVG_PICK_STROKE : NVG_PICK_FILL);
```

Comment:
It was originally for supporting holes interacting with strokes, but then I noticed nanovg didn't seems to handle that to I removed that code but left this. I'll remove references to it.